src/System.Net.Http/src/System/Net/Http/StreamToStreamCopy.cs

```diff
@@ -8,69 +8,52 @@
 namespace System.Net.Http
 {
     // This helper class is used to copy the content of a source stream to a destination stream.
-    // The type verifies if the source and/or destination stream are MemoryStreams (or derived types). If so, sync 
+    // The type verifies if the source and/or destination stream are MemoryStreams (or derived types). If so, sync
     // read/write is used on the MemoryStream to avoid context switches.
-    internal class StreamToStreamCopy
+    internal static class StreamToStreamCopy
     {
-        private byte[] _buffer;
-        private int _bufferSize;
-        private Stream _source;
-        private Stream _destination;
-        private bool _disposeSource;
-        private bool _sourceIsMemoryStream;
-        private bool _destinationIsMemoryStream;
-
-        public StreamToStreamCopy(Stream source, Stream destination, int bufferSize, bool disposeSource)
+        public static async Task CopyAsync(Stream source, Stream destination, int bufferSize, bool disposeSource)
         {
             Contract.Requires(source != null);
             Contract.Requires(destination != null);
             Contract.Requires(bufferSize > 0);
 
-            _buffer = new byte[bufferSize];
-            _source = source;
-            _destination = destination;
-            _bufferSize = bufferSize;
-            _disposeSource = disposeSource;
-            _sourceIsMemoryStream = source is MemoryStream;
-            _destinationIsMemoryStream = destination is MemoryStream;
-        }
+            byte[] buffer = new byte[bufferSize];
 
-        public async Task StartAsync()
-        {
             // If both streams are MemoryStreams, just copy the whole content at once to avoid context switches.
             // This will not block since it will just result in a memcopy.
-            if (_sourceIsMemoryStream && _destinationIsMemoryStream)
+            if (source is MemoryStream && destination is MemoryStream)
```

Comment:
I'm in the process of working on a change to HttpContent (the base class of these *Content classes) to use TryGetBuffer() and expose that internally in the class in order to optimize/fix some scenarios in the .NETNative version of this library.